src/Management/src/Endpoint/HeapDump/HeapDumpEndpointMiddleware.cs

```diff
@@ -10,55 +11,44 @@
 
 namespace Steeltoe.Management.Endpoint.HeapDump;
 
-public class HeapDumpEndpointMiddleware : EndpointMiddleware<string>
+internal sealed class HeapDumpEndpointMiddleware : EndpointMiddleware<object, string>
 {
-    public HeapDumpEndpointMiddleware(IHeapDumpEndpoint endpoint, IOptionsMonitor<ManagementEndpointOptions> managementOptions,
-        ILogger<HeapDumpEndpointMiddleware> logger)
-        : base(endpoint, managementOptions, logger)
+    private readonly ILogger<HeapDumpEndpointMiddleware> _logger;
+
+    public HeapDumpEndpointMiddleware(IHeapDumpEndpointHandler endpointHandler, IOptionsMonitor<ManagementEndpointOptions> managementOptions,
+        ILoggerFactory loggerFactory)
+        : base(endpointHandler, managementOptions, loggerFactory)
     {
-        Endpoint = endpoint;
+        _logger = loggerFactory.CreateLogger<HeapDumpEndpointMiddleware>();
     }
 
-    public override Task InvokeAsync(HttpContext context, RequestDelegate next)
+    protected override async Task<string> InvokeEndpointHandlerAsync(HttpContext context, CancellationToken cancellationToken)
     {
-        if (Endpoint.Options.ShouldInvoke(managementOptions, context, logger))
-        {
-            return HandleHeapDumpRequestAsync(context);
-        }
-
-        return Task.CompletedTask;
+        return await EndpointHandler.InvokeAsync(null, context.RequestAborted);
     }
-
-    protected internal async Task HandleHeapDumpRequestAsync(HttpContext context)
+    protected override async Task WriteResponseAsync(string fileName, HttpContext context, CancellationToken cancellationToken)
     {
-        string fileName = Endpoint.Invoke();
-        logger.LogDebug("Returning: {fileName}", fileName);
-        context.Response.Headers.Add("Content-Type", "application/octet-stream");
+        _logger.LogDebug("Returning: {fileName}", fileName);
 
         if (!File.Exists(fileName))
         {
             context.Response.StatusCode = StatusCodes.Status404NotFound;
             return;
         }
 
-        string gzFileName = $"{fileName}.gz";
-        Stream result = await Utils.CompressFileAsync(fileName, gzFileName, logger);
+        context.Response.ContentType = "application/octet-stream";
+        context.Response.Headers.Add("Content-Disposition", $"attachment; filename=\"{Path.GetFileName(fileName)}\"");
```

Comment:
This wasn't correctly taken from https://github.com/SteeltoeOSS/Steeltoe/pull/1130#discussion_r1227893083. The '.gz' file extension is missing (original at https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Management/src/Endpoint/HeapDump/HeapDumpEndpointMiddleware.cs#L44)

```suggestion
        context.Response.Headers.Add("Content-Disposition", $"attachment; filename=\"{Path.GetFileName(fileName)}.gz\"");
```